From 150ccfe3340e485fc7105ea55b38fbba7472c33c Mon Sep 17 00:00:00 2001
From: Junping Du <junping_du@apache.org>
Date: Wed, 30 Aug 2017 13:30:14 -0700
Subject: [PATCH 2694/2748] HADOOP-14814. Fix incompatible API change on
 FsServerDefaults to HADOOP-14104. Contributed by
 Junping Du.

(cherry picked from commit 41480233a9cfb0bcfb69cc0f1594120e7656f031)
(cherry picked from commit 7dd5bb949206fb635d875cd651292b451267a951)
(cherry picked from commit f9ebcebe74201b1c3997269927dfade50d89a201)

 Conflicts:
	hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsServerDefaults.java

Change-Id: I91840e81ac04d9e5444ebaae206114649a8b0725
---
 .../org/apache/hadoop/fs/FsServerDefaults.java     |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsServerDefaults.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsServerDefaults.java
index 469243c..52d445e 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsServerDefaults.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsServerDefaults.java
@@ -62,6 +62,15 @@ public FsServerDefaults() {
   public FsServerDefaults(long blockSize, int bytesPerChecksum,
       int writePacketSize, short replication, int fileBufferSize,
       boolean encryptDataTransfer, long trashInterval,
+      DataChecksum.Type checksumType) {
+    this(blockSize, bytesPerChecksum, writePacketSize, replication,
+        fileBufferSize, encryptDataTransfer, trashInterval, checksumType,
+        null);
+  }
+
+  public FsServerDefaults(long blockSize, int bytesPerChecksum,
+      int writePacketSize, short replication, int fileBufferSize,
+      boolean encryptDataTransfer, long trashInterval,
       DataChecksum.Type checksumType,
       String keyProviderUri) {
     this.blockSize = blockSize;
-- 
1.7.9.5

